### ISSUE TITLE
'tomlkit.items.Key.__eq__()' should return 'NotImplemented' when comparing objects without a 'key' attribute

### ISSUE THREAD

**Issue by cosmicspacedragon** [sdispater/tomlkit#9]
The current comparison method does not take into account the possibility that an object without a `key` attribute could be passed in as `other`.

What follows is some slightly obfuscated output from one of my pytest runs.
```
self = <Key "a\lovely\key">
other = 'a\\lovely\\key'

    def __eq__(self, other):  # type: (Key) -> bool
>       return self.key == other.key
E       AttributeError: 'str' object has no attribute 'key'

..\..\.virtualenvs\<virtualenv>\lib\site-packages\tomlkit\items.py:164: AttributeError
```
As you can see, I tried comparing a 'tomlkit.items.Key' against a regular string which failed spectacularly.
If I'm not mistaken, the worst that should happen when doing a comparison is getting `False`.

**Comment by sdispater** [comment]
I think being able to compare a `Key` to a string is reasonable enough and more intuitive when working with a document.

I agree however that for other types, `__eq__()` should return `NotImplemented`.

**Comment by sdispater** [comment]
This is fixed in the latest release (`0.4.5`).
